pkg/config/environment_containers.go

```diff
@@ -0,0 +1,163 @@
+	KubeOrchestratorExplorer Feature = "orchestratorExplorer"
+
+	defaultLinuxDockerSocket       = "/var/run/docker.sock"
+	defaultWindowsDockerSocketPath = "//./pipe/docker_engine"
+	defaultLinuxContainerdSocket   = "/var/run/containerd/containerd.sock"
+	defaultLinuxCrioSocket         = "/var/run/crio/crio.sock"
+	defaultHostMountPrefix         = "/host"
+	unixSocketPrefix               = "unix://"
+	winNamedPipePrefix             = "npipe://"
+
+	socketTimeout = 500 * time.Millisecond
+)
+
+func detectContainerFeatures(features FeatureMap) {
+	detectKubernetes(features)
+	detectDocker(features)
+	detectContainerd(features)
+	detectFargate(features)
+}
+
+func detectKubernetes(features FeatureMap) {
+	if IsKubernetes() {
+		features[Kubernetes] = struct{}{}
+		if Datadog.GetBool("orchestrator_explorer.enabled") {
+			features[KubeOrchestratorExplorer] = struct{}{}
+		}
+	}
+}
+
+func detectDocker(features FeatureMap) {
+	if _, dockerHostSet := os.LookupEnv("DOCKER_HOST"); dockerHostSet {
+		features[Docker] = struct{}{}
+	} else {
+		for _, defaultDockerSocketPath := range getDefaultDockerPaths() {
+			exists, reachable := system.CheckSocketAvailable(defaultDockerSocketPath, socketTimeout)
+			if exists && !reachable {
+				log.Infof("Agent found Docker socket at: %s but socket not reachable (permissions?)", defaultDockerSocketPath)
+			}
+
+			if exists && reachable {
+				features[Docker] = struct{}{}
+
+				// Even though it does not modify configuration, using the OverrideFunc mechanism for uniformity
+				AddOverrideFunc(func(Config) {
+					os.Setenv("DOCKER_HOST", getDefaultDockerSocketType()+defaultDockerSocketPath)
+				})
+				break
+			}
+		}
+	}
+}
+
+func detectContainerd(features FeatureMap) {
+	// CRI Socket - Do not automatically default socket path if the Agent runs in Docker
+	// as we'll very likely discover the containerd instance wrapped by Docker.
+	criSocket := Datadog.GetString("cri_socket_path")
+	if criSocket == "" && !IsDockerRuntime() {
+		for _, defaultCriPath := range getDefaultCriPaths() {
+			exists, reachable := system.CheckSocketAvailable(defaultCriPath, socketTimeout)
+			if exists && !reachable {
+				log.Infof("Agent found cri socket at: %s but socket not reachable (permissions?)", defaultCriPath)
+			}
+
+			if exists && reachable {
+				criSocket = defaultCriPath
+				AddOverride("cri_socket_path", defaultCriPath)
+				// Currently we do not support multiple CRI paths
+				break
+			}
+		}
+	}
+
+	if criSocket != "" {
+		// Containerd support was historically meant for K8S
+		// However, containerd is now used standalone elsewhere.
+		// TODO: Consider having a dedicated setting for containerd standalone
+		if IsKubernetes() {
+			features[Cri] = struct{}{}
+		}
+
+		if strings.Contains(criSocket, "containerd") {
+			features[Containerd] = struct{}{}
+		}
+	}
+}
+
+func detectFargate(features FeatureMap) {
+	if IsECSFargate() {
+		features[ECSFargate] = struct{}{}
+	}
+
+	if Datadog.GetBool("eks_fargate") {
```

Comment:
you can use the `IsEKSFargateInstance()` function from `pkg/util/fargate`

Reply:
It would create a circular dependency unfortunately